### PR TITLE
Iphlpapi addition

### PR DESCRIPTION
**Pull request by BlackHoleFox**
Ended up needing these APIs for a project and at Pete's request created this PR based off another PR that had already made the needed additions. All credit go to PR's original author @mattlknight in #675. All the tests pass as well.

Review away :) @retep998

### CODE REVIEW

**Comment by mattlknight** [comment]
@retep998 @BlackHoleFox , this PR seems much cleaner and better than mine lol. @BlackHoleFox do you want to compare with mine to see if there are any differences? I'm trying to figure out how to do that myself right now.

**Comment by mattlknight** [comment]
@BlackHoleFox oh jeeze, I can already see files in your branch that appear to be missing from mine somehow. I feel like I've redone the same work 2-3 times now lol. One of these days I'll figure out github, forks, and merges.

**Comment by mattlknight** [comment]
@BlackHoleFox well, now I'm lost, when I do a GitHub compare across forks, and I compare your 0.3 branch with my 0.3 branch, it looks like I'm missing everything you have. But when I look at the files manually, we have the same content. So I must be doing compares incorrectly.

File: src/shared/ipifcons.rs [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+pub const IF_TYPE_IEEE802154: IFTYPE = 259;
+pub const IF_TYPE_XBOX_WIRELESS: IFTYPE = 281;
+pub const MAX_IF_TYPE: IFTYPE = 281;
+ENUM!{enum IF_ACCESS_TYPE {
+    IF_ACCESS_LOOPBACK             = 1,
+    IF_ACCESS_BROADCAST            = 2,
+    IF_ACCESS_POINT_TO_POINT       = 3,
+    IF_ACCESS_POINTTOPOINT         = 3,
+    IF_ACCESS_POINT_TO_MULTI_POINT = 4,
+    IF_ACCESS_POINTTOMULTIPOINT    = 4,
```
Comment (retep998) [inline src/shared/ipifcons.rs]: No aligned indentation.

File: src/um/iptypes.rs [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+pub const MAX_HOSTNAME_LEN: usize               = 128;
+pub const MAX_DOMAIN_NAME_LEN: usize            = 128;
+pub const MAX_SCOPE_ID_LEN: usize               = 256;
+pub const MAX_DHCPV6_DUID_LENGTH: usize         = 130;
+pub const MAX_DNS_SUFFIX_STRING_LENGTH: usize   = 256;
+pub const BROADCAST_NODETYPE: usize    = 1;
+pub const PEER_TO_PEER_NODETYPE: usize = 2;
+pub const MIXED_NODETYPE: usize        = 4;
+pub const HYBRID_NODETYPE: usize       = 8;
+pub type __time64_t = LONGLONG;
```
Comment (retep998) [inline src/um/iptypes.rs]: I can't find this definition in this header.

**Review by retep998 — CHANGES_REQUESTED** [review]
There's a lot of evil aligned indentation to be fixed.

**Comment by BlackHoleFox** [comment]
Fixed the indentation issues throughout the PR and about the unknown definitions, they do seem to exist according to https://docs.microsoft.com/en-us/windows/desktop/api/iphlpapi/nf-iphlpapi-getnetworkparams

File: src/ucrt/corecrt.rs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// Licensed under the Apache License, Version 2.0
+// <LICENSE-APACHE or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// All files in the project carrying such notice may not be copied, modified, or distributed
+// except according to those terms.
+// #include <time.h>
+// #include <vcruntime.h>
+use shared::ntdef::{CHAR, LONGLONG, PCHAR, PWCHAR, ULONG, ULONGLONG, WCHAR};
+pub type __time32_t = LONG;
+pub type __time64_t = LONGLONG;
```
Comment (retep998) [inline src/ucrt/corecrt.rs]: These definitions need to match the ones in the headers. `long` -> `c_long` and `__int64` -> `__int64`.

File: src/ucrt/mod.rs [hunk 4]
```diff
@@ -0,0 +1,7 @@
+// Licensed under the Apache License, Version 2.0
+// <LICENSE-APACHE or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// All files in the project carrying such notice may not be copied, modified, or distributed
+// except according to those terms.
+// Headers for the universal C Runtime
+#[cfg(feature = "corecrt")] pub mod corecrt;
```
Comment (retep998) [inline src/ucrt/mod.rs]: Don't forget the trailing newline

File: src/ucrt/corecrt.rs (hunk 3 above)
Comment (BlackHoleFox) [reply src/ucrt/corecrt.rs]: Fixed in the latest commit.

File: src/ucrt/mod.rs (hunk 4 above)
Comment (BlackHoleFox) [reply src/ucrt/mod.rs]: Fixed as well, thanks for the reminder.

**Comment by mattlknight** [comment]
@retep998 Can this be merged?

**Comment by retep998** [comment]
Why do you have commits by me in this PR?

**Pull request closed without merging**